test/VarianceTest.cc

```diff
@@ -341,19 +342,141 @@ INSTANTIATE_TEST_CASE_P(
-        VarianceParam(64, 32, &eb_aom_variance64x32_c, &eb_aom_variance64x32_avx2),
-        VarianceParam(64, 64, &eb_aom_variance64x64_c, &eb_aom_variance64x64_avx2),
-        VarianceParam(64, 128, &eb_aom_variance64x128_c, &eb_aom_variance64x128_avx2),
-        VarianceParam(128, 64, &eb_aom_variance128x64_c, &eb_aom_variance128x64_avx2),
+        VarianceParam(32, 16, &eb_aom_variance32x16_c,
+                      &eb_aom_variance32x16_avx2),
+        VarianceParam(32, 32, &eb_aom_variance32x32_c,
+                      &eb_aom_variance32x32_avx2),
+        VarianceParam(32, 64, &eb_aom_variance32x64_c,
+                      &eb_aom_variance32x64_avx2),
+        VarianceParam(64, 16, &eb_aom_variance64x16_c,
+                      &eb_aom_variance64x16_avx2),
+        VarianceParam(64, 32, &eb_aom_variance64x32_c,
+                      &eb_aom_variance64x32_avx2),
+        VarianceParam(64, 64, &eb_aom_variance64x64_c,
+                      &eb_aom_variance64x64_avx2),
+        VarianceParam(64, 128, &eb_aom_variance64x128_c,
+                      &eb_aom_variance64x128_avx2),
+        VarianceParam(128, 64, &eb_aom_variance128x64_c,
+                      &eb_aom_variance128x64_avx2),
         VarianceParam(128, 128, &eb_aom_variance128x128_c,
                       &eb_aom_variance128x128_avx2)));
 
-}  // namespace
+typedef std::tuple<int, int> BlockSize;
+BlockSize TEST_BLOCK_SIZES[] = {
+    BlockSize(4, 4),    BlockSize(4, 8),    BlockSize(4, 16),
+    BlockSize(8, 4),    BlockSize(8, 8),    BlockSize(8, 16),
+    BlockSize(8, 32),   BlockSize(16, 4),   BlockSize(16, 8),
+    BlockSize(16, 16),  BlockSize(16, 32),  BlockSize(16, 64),
+    BlockSize(32, 8),   BlockSize(32, 16),  BlockSize(32, 32),
+    BlockSize(32, 64),  BlockSize(64, 16),  BlockSize(64, 32),
+    BlockSize(64, 64),  BlockSize(64, 128), BlockSize(128, 64),
+    BlockSize(128, 128)};
+/**
+ * @brief Unit test for variance functions, target functions include:
+ *  - highbd_variance64_{c,avx2}
+ *
+ * Test strategy:
+ *  This test case contains zero test, random value test, one quarter test as
+ * test pattern.
+ *
+ *
+ * Expect result:
+ *  Results come from reference functon and target function are
+ * equal.
+ *
+ * Test coverage:
+ *
+ * Test cases:
+ *
+ */
+class HBDVarianceTest : public ::testing::TestWithParam<BlockSize> {
+  public:
+    HBDVarianceTest() : width_(TEST_GET_PARAM(0)), height_(TEST_GET_PARAM(1)) {
+        src_data_ =
+            reinterpret_cast<uint8_t *>(eb_aom_memalign(32, MAX_BLOCK_SIZE));
+        ref_data_ =
+            reinterpret_cast<uint8_t *>(eb_aom_memalign(32, MAX_BLOCK_SIZE));
+    }
+
+    ~HBDVarianceTest() {
+        eb_aom_free(src_data_);
+        src_data_ = nullptr;
+        eb_aom_free(ref_data_);
+        ref_data_ = nullptr;
+    }
+
+    void run_zero_test() {
+        for (int i = 0; i < 256; ++i) {
+            memset(src_data_, i, MAX_BLOCK_SIZE);
+            memset(ref_data_, i, MAX_BLOCK_SIZE);
+        }
```

Comment:
Why do we repeat memset with different values? should it be 
```
memset(src_data_, 0, MAX_BLOCK_SIZE);
memset(ref_data_, 0, MAX_BLOCK_SIZE);
```

Reply:
modified.